core/domain/exp_jobs_one_off_test.py

```diff
@@ -43,6 +47,59 @@ def mock_get_filename_with_dimensions(filename, unused_exp_id):
         filename, 490, 120)
 
 
+def mock_save_original_and_compressed_versions_of_image(
+        user_id, filename, exp_id, original_image_content):
+    filepath = 'image/%s' % filename
+
+    filename_wo_filetype = filename[:filename.rfind('.')]
+    filetype = filename[filename.rfind('.') + 1:]
+
+    compressed_image_filename = '%s_compressed.%s' % (
+        filename_wo_filetype, filetype)
+    compressed_image_filepath = 'image/%s' % compressed_image_filename
+
+    micro_image_filename = '%s_micro.%s' % (
+        filename_wo_filetype, filetype)
+    micro_image_filepath = 'image/%s' % micro_image_filename
+
+    fs = fs_domain.AbstractFileSystem(fs_domain.GcsFileSystem(
+        'exploration/%s' % exp_id))
+
+    if not fs.isfile(filepath.encode('utf-8')):
+        fs.commit(
+            user_id, filepath.encode('utf-8'), original_image_content,
+            mimetype='image/%s' % filetype)
+
+    if not fs.isfile(compressed_image_filepath.encode('utf-8')):
+        fs.commit(
+            user_id, compressed_image_filepath.encode('utf-8'),
+            original_image_content, mimetype='image/%s' % filetype)
+
+    if not fs.isfile(micro_image_filepath.encode('utf-8')):
+        fs.commit(
+            user_id, micro_image_filepath.encode('utf-8'),
+            original_image_content, mimetype='image/%s' % filetype)
+
+
+def run_job_for_deleted_exp(
+        self, job_class, check_error=False,
+        error_type=None, error_msg=None, function_to_be_called=None,
+        exp_id=None):
+    """Helper function to run job for an deleted exploration and check the
+    output or error condition.
+    """
+    job_id = job_class.create_new()
+    job_class.enqueue(job_id)
```

Comment:
Before and after this statement, can we assert the number of jobs running in the task queue? This is just to ensure that the job being run is the one in question, and not any other stray process.

Reply:
Done.